src/Sarif/Baseline/ResultMatching/SarifLogResultMatcher.cs

```diff
@@ -278,6 +278,10 @@ private List<ExtractedResult> ExtractResultsFromRuns(IEnumerable<Run> sarifRuns)
             run.Graphs = graphs;
             run.Invocations = invocations;
             run.Properties = properties;
+            run.VersionControlProvenance = MergeList(
+                currentRuns?.SelectMany(run => run.VersionControlProvenance ?? Enumerable.Empty<VersionControlDetails>()),
```

Comment:
>currentRuns [](http://example.com/codeflow?start=16&length=11)

this is a poor practice, to iterate over these collections and the pass that result to a helper that will iterate again. please rewrite to avoid that. looks like you already provide handling in the callee for the null run.VersionControlProvenance case.

Reply:
looks like you already provide handling in the callee for the null run.VersionControlProvenance case.  
  ------  Just to comment on this line, with the current code if there is one run.VersionControlProvenance is null, the C# SelectMany() will have NullRef exception. 

This is by design of the SelectMany() when any one of the selector returns null.

This can not be handled by null check inside MergeList() method, because the error happens at SelectMany().

for SelectMany people usually write this way:
`AAAA
.Where(x => x.property1 != null)
.SelectMany(x => x.property1)`

(Just for discussion don't make changes according to my comment :) )

Reply:
refactor this function avoid the duplicated iteration of the collection, also no need to handle the null value in the enumerable the old function has to deal with. 
please see the latest change.